Add Original Remap Palette for RA Campaign

MustaphaTR opened the pull request:
,Co-Op Missions and Shellmap.

MustaphaTR committed:
Add Original Remap Palette for RA Mod Campaign

and Shellmap

Arular101 reviewed (approved):
👍

I really like this change. It brings the colors back to how the original game was.

I tested the campaign and saw that in some missions the colors aren't right. However, this has noting to do with this PR.

![palette 1](https://user-images.githubusercontent.com/33390659/32680755-74d506e8-c66c-11e7-8373-d5d4bad67ad9.png)

Soviets 02a and b
![palette s2](https://user-images.githubusercontent.com/33390659/32681460-e671a542-c66f-11e7-8c3d-f30721817f97.png)

Soviets 03
![palette s3](https://user-images.githubusercontent.com/33390659/32681465-edbce7e4-c66f-11e7-8bbb-fdbcff8e8677.png)

Soviets 05
![palette 2](https://user-images.githubusercontent.com/33390659/32680793-af19c3ca-c66c-11e7-92fa-0de597ec1d21.png)

These changes should be made:
Allies 02: convoy trucks.
Allies 03b: ranger.
Soviets 02a and b: enemy color.
Soviets 03: enemy and own color.
Soviets 05: some enemy units and the base that gets build on the island.

MustaphaTR commented:
My RA1 installation didn't have missions in it, so i'm installing currently and will test the stuff to be sure.

> Allies 03b: ranger.

I'm not sure what exactly going on there, ranger is owned by neutral on OpenRA, i'll check how was it in original.

> Allies 02: convoy trucks.
> Soviets 02a and b: enemy color.
> Soviets 03: enemy 

Either WW has colors changes specifically for those missions, or someone messed the faction names during conversion, will check.

> Soviets 03: own color.
> Soviets 05: some enemy units and the base that gets build on the island.

These are my fault, i forgot to add colors of GoodGuy and BadGuy player names.

> Soviets 05: Radar Dome

Radar is owned by Greece in this mission for OpenRA, probably changed so it won't show black low power overlay. Some .yaml editing can fix the problem.

MustaphaTR commented:
Updated.

> Allies 02: convoy trucks.

Replaced France with England, which fixes the colors.

> Allies 03b: ranger.

Added an England player and given the Ranger to it.

> Soviets 02a and b: enemy color.
> Soviets 03: enemy

Replaced Germany with Greece, which fixes the colors.

> Soviets 03: own color.

Renamed Soviet to BadGuy, and Added BadGuy to `campaign-palettes.yaml`.

> Soviets 05: some enemy units and the base that gets build on the island.

Added GoodGuy to `campaign-palettes.yaml`.

> Soviets 05: Radar Dome

Added a France player and given Radar Dome to it. ~~Removed `WithColoredOverlay@IDISABLE:` from Radar Dome.~~

Edit: I just realised what i did to Radar Dome low power overlay is not really a good way (as Player can also build a Radar Dome), i'll try something else.

Arular101 commented:
Oh sorry, I forgot to mention that the radar dome in Soviets 05 could stay blue. In my opinion that is fine. I let it up to you if you want to change it to green or let it stay blue.

MustaphaTR commented:
I already changed it to French colors and is working fine. I'm just unsure if my approach is correct.

Arular101 commented:
Again great job at fixing it so fast! I tested the above mentioned changes and all are good now. I also tested the radar dome in Soviet 05 and the only thing I can say is that it works function wise. I can't say if the approach is good or not.

I tested the rest of the mission and all looks good, but I have to say I only tested the beginnings of these missions. That's because, I haven't played these missions yet and I want to play co-op with a friend without knowing the mission. The thing what I'm not sure is the third Allied faction in the missions Evacuation, Exodus and Infiltration.
![palette 5](https://user-images.githubusercontent.com/33390659/32700244-8a088b50-c7c2-11e7-88f7-5ece7b527aa2.png)

Only the co-op person in Evacuation, Exodus and Infiltration and survival 01 and 02 are a slightly different green. To me it is fine, so 👍 and it's got to be said,  the green in Evacuation, Exodus and Infiltration is really nice!

MustaphaTR commented:
Monster Tank Madness is a mission from Aftermath expansion. All other coop and survival missions were probably also trying to imitate the original colors, so changes to these colors are what should happen.

Arular101 commented:
> Monster Tank Madness is a mission from Aftermath expansion.

Yup, this one is 100% correct.

pchote commented:
Needs a rebase.

Arular101 commented on mods/ra/maps/soviet-05/rules.yaml:
```diff
@@ -84,7 +84,14 @@ APC:
+		RequiresCondition: !french
+	Targetable@NOAUTOTARGET:
+		RequiresCondition: french
+		TargetTypes: Ground, Structure, C4, DetonateAttack, SpyInfiltrate, NoAutoTarget
```
The targetable traits can be removed now. This is not needed anymore since the current release doesn't automatically  target buildings.

MustaphaTR replied:
I think it is better to keep Radar non-auto targeted even on Attack-Anything stance here, which is what this does.

Arular101 replied:
I understand you. The only thing is, this is the only building with such a rule in the campaign/skirmish.

abcdefg30 replied:
We might want to change the other buildings as well then.

Arular101 replied:
Wouldn't that be just an error of the player self by choosing attack anything while there is an important building? Just a thought, I won't object it.

pchote replied:
I agree with @Arular101 here: so long as we we properly and promptly fail the mission (as opposed to leaving it in a state where the game continues but is unwinnable) then it really falls on the player to not put their units into the non-default state where they will fail the mission for you.

Arular101 reviewed (approved):
👍 Perfect!

I just would like to know if there are intentions to bring the original palette to the rest of the game.

GraionDilach commented:
I wouldn't bring the original colors to skirmish/multiplayer - there were only 8 colors afterall and we ship maps with 12 players.

Arular101 commented:
Oh yeah, I was thinking about the same palettes will be used for all the colors (the buildings look a little bit shinier, most noticeable on the roof of the war factory), but I guess that is not easy to implement.

pchote commented:
> Either WW has colors changes specifically for those missions, or someone messed the faction names up during conversion, will check.

You replaced the factions in our versions, but didn't confirm which of these was the case.  I'm assuming it was the latter?

MustaphaTR commented:
Yes, latter. Whoever made the missions for OpenRA used different names for factions than original.

reaperrr commented:
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/8dc3b6d9bc39974bb49247c18dcedfb140b28994...b44c03487b733e61d79bd3ce1fea1a81dc24646c)
